schemes/enc/v1/ciphers.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2023 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1
+
+import (
+	"errors"
+	"strconv"
+)
+
+// Cipher used to encrypt the file.
+type Cipher string
+
+const (
+	CipherAESGCM           Cipher = "AES-GCM"
+	CipherChaCha20Poly1305 Cipher = "CHACHA20-POLY1305"
+
+	cipherNumAESGCM = 1
+	cipherNumChaCha20Poly1305 = 2
+)
+
+// Validate the passed cipher and resolves aliases.
+func (a Cipher) Validate() (Cipher, error) {
+	switch a {
+	// Valid ciphers, not aliased
+	case CipherAESGCM, CipherChaCha20Poly1305:
+		return a, nil
+
+	default:
+		return a, errors.New("cipher " + string(a) + " is not supported")
```

Comment:
Could we use `fmt` rather than `+`?

Reply:
Sure, it's an error here anyways so (hopefully) not a hot path